### PR TITLE
MQ ER

### PR DESCRIPTION
**Pull request by rattus128**
Child dungeons, fire and GTG. Rest to come

### CODE REVIEW

**Comment by r0bd0g** [comment]
Did you check if the bombs reach the boulders in MQ deku without song of time as adult?
I think I personally would have put going to the water room in reverse in logic. The door reclosing is pretty dumb though lol.
If some combo of sticks and explosives is okay for the spirit temple iron knuckles, then something similar should be okay for the gtg one. Did you know that after the armor is off, the slingshot even works?
Why do hover boots work to reach second room upper chest? You'd think if you couldn't get there like that in vanilla it wouldn't work in mq either? Anyway, one also figures you should be able to hookshot the chest? But maybe there isn't a line of sight, I dunno.
Why'd you remove the Kokiri Sword check from big octo? Actually I bet adult could kill that boss for you...

I still find the logic really hard to follow at times. as adult is adult as adult here... dizzy now.

**Comment by Roman971** [comment]
The nut pot location shouldn't be in the eventlocations dict because that location doesn't exist in vanilla Jabu so seeds with MQ Jabu won't be able to generate. It should probably be handled like similar dungeon event locations are.

```json
"Dodongos Cavern Bomb Bag Area": "is_adult or (as_adult(True) and has_explosives)",
```
I'm not sure what's going on there, didn't you mean to use `as_adult_here` instead of `as_adult(True)` ?

Also, there are a few `can_use(bow)` in the DC MQ logic, I think they should be `can_use(Bow)`.

**Comment by rattus128** [comment]
@Roman971 . All fixed. Thanks.

**Comment by rattus128** [comment]
@r0bd0g ,
Bombs are an easy throw in MQ Deku Compass GS, so added added.

Fire temple like like sling added

There is a timer for the iron knuckle. I got the kill kiting with bombs only. It's a little jank with z targetting in the quicksand but once you get used to it its not terrible. Its actually similarly difficult to whaling on him with the kokiri sword from behind in the quicksand. So I added bombs to match vanilla spirits has_explosives condition. I want to leave sticks and sling out as I think you would need consistent JS fighting to ammo manage sticks.

Jabu hovers do work in vanilla, the problem is you cant roll in the air which means you cant carry the crate over the the switch. So despite being able to get to the ledge adult is totally useless in vanilla Jabu. The hookshot works on the MQ chest, so added that.

Think I removed that KS by accident refactoring the regions, will look into a better organisation to restore that. Otherwise all fixed. Dizzy myself, too much code for today.

**Comment by r0bd0g** [comment]
Sounds like that trick in jabu to roll from the switch as it closes will be logically relevant in ER...

**Comment by r0bd0g** [comment]
Jabu:
Still a Kokiri Sword check on Jabu Boss Area seems to be missing?

I wonder if Adult can get the invisible enemies GS in some way, without having to deal with the invisible enemies? This could allow a lens requirement to be skipped, I guess as long as the invisible enemies are not too intrusive as you attempt to do it.

GTG:
I still feel like sticks+slingshot should be fair enough on the Iron Knuckles? 5 jumpslashes… Would anybody be crazy enough to not know to use jumpslashes? I dunno, it's close.

Water:
Isn't the Hookshot required actually to spawn the Map chest? And hook is needed for an underwater switch for central pillar chest as well. TBH we lucked out that we don't have to discuss whether the janky jump should be in logic or not.

Can you explain the as_child route for the Water Compass Chest?

Ice:
Is child tall enough to through the pot at the switches? As adult I had to pre-destroy an icicle before making the run with a pot. Can child slip by without needing to do that?

Doesn't the Map Chest require blue fire? Is it possible to use a pot on the switch that spawns the chest?

Forest:
What is the status of Deku Baba Nuts/Sticks these days? At this point I'm confused as to how it's supposed to work now with full ER.

GS Forest Temple MQ Block Push Room - Why only Kokiri Sword for this?

GS Forest Temple MQ Outdoor West - Isn't that GS just sitting there? Why Hookshot or Rang?

Is it possible to access the outdoor ledge from the block room in some way as child? You'll probably need Strength to do it at the least. And maybe adult hits the switch or maybe child can jumpslash it as well I dunno? And then maybe cross through the well somehow, perhaps pre-lowering it as Adult or using Gold Scale? I think it might be possible to Boomerang the Outdoor East GS without needing Slingshot to access that courtyard. But I don't know if any of what I suggest here is actually possible.

**Comment by rattus128** [comment]
> Jabu:
> Still a Kokiri Sword check on Jabu Boss Area seems to be missing?
> 

Fixed

> I wonder if Adult can get the invisible enemies GS in some way, without having to deal with the invisible enemies? This could allow a lens requirement to be skipped, I guess as long as the invisible enemies are not too intrusive as you attempt to do it.
> 

Hovers added

> GTG:
> I still feel like sticks+slingshot should be fair enough on the Iron Knuckles? 5 jumpslashes… Would anybody be crazy enough to not know to use jumpslashes? I dunno, it's close.
> 
Jump slashes are crippled somewhat by the quicksand making them a little jank. The timer is too tight to kite him to solid ground, so i'm erring on out. The "beginner friendly" strat (if there even is such a thing with this guy) is going to be to get behind him and stay behind him with KS PCS.

> Water:
> Isn't the Hookshot required actually to spawn the Map chest? And hook is needed for an underwater switch for central pillar chest as well. TBH we lucked out that we don't have to discuss whether the janky jump should be in logic or not.
> 
Correct on all. I deleted that comment.

> Can you explain the as_child route for the Water Compass Chest?
> 
Adult drains but has no bow or fire. Child returns with sticks to light torches (NOTE: non permanent). The only way child can kill the spike is explosives (srsly this game ...). Sticks kill Lizalfos. The reachability is done on the Lobby rather than as_child_here() because child is shut out of the dungeon on the dive exit logic (Irons). Its easier to leave child detached like this and cherry-pick the few (only one in MQ) cases where child assist adult in WT.

> Ice:
> Is child tall enough to through the pot at the switches?

Yes, its not too precise either. Its less precise than the high bomb throw in Jabu MQ.

> As adult I had to pre-destroy an icicle before making the run with a pot. Can child slip by without needing to do that?

Yes, slip past.

> 
> Doesn't the Map Chest require blue fire? Is it possible to use a pot on the switch that spawns the chest?
> 

Fixed

> Forest:
> What is the status of Deku Baba Nuts/Sticks these days? At this point I'm confused as to how it's supposed to work now with full ER.
> 
Fixed for the MQ dungeon. We need to look harder at the Kokiri Forest though.

> GS Forest Temple MQ Block Push Room - Why only Kokiri Sword for this?
> 

Stalfos in the connector room. I guest cause there's only the one, sticks and nuts is reasonable. Your call. I was going for consistency with vanilla where the Stalfos pairs require KS. I would avoid just sticks due to shielded mishits.

> GS Forest Temple MQ Outdoor West - Isn't that GS just sitting there? Why Hookshot or Rang?
> 

Yup. Fixed. That was me being a derp and confusing it with the East.

> Is it possible to access the outdoor ledge from the block room in some way as child? You'll probably need Strength to do it at the least. And maybe adult hits the switch or maybe child can jumpslash it as well I dunno? 

Even if I chu and adult-JS both switches I think I'm still stuck as child. Child is too short to climb the push blocks, the glass platform shortcut to up top requires hook and the lower door to the ledge requires hovers.

Thanks as always.

> And then maybe cross through the well somehow, perhaps pre-lowering it as Adult or using Gold Scale? I think it might be possible to Boomerang the Outdoor East GS without needing Slingshot to access that courtyard. But I don't know if any of what I suggest here is actually possible.

**Comment by rattus128** [comment]
@r0bd0g , one thing I forgot above, in ice if you use dins to get to the map room, you can't use pots strats on the switch because there is a row of stalactites you can't pass carrying a pot.

**Comment by r0bd0g** [comment]
There's another single stalfos in mq spirit that's sword or sticks I think on 4.0. It's a bit different b/c you could actually push him into the pit to win, maybe with shield or nuts or just get lucky. So I didn't really know what to do with that guy either, lol.

**Comment by r0bd0g** [comment]
Iron Boots for the Forest Well GS is going to need Hookshot to kill it with.

**Comment by rattus128** [comment]
> Iron Boots for the Forest Well GS is going to need Hookshot to kill it with.

Fixed
